WordPress/Classes/Services/MediaService.m

```diff
@@ -269,6 +269,36 @@ - (void)updateMedia:(Media *)media
                 failure:failureBlock];
 }
 
+- (void)updateMultipleMedia:(NSArray<Media *> *)mediaObjects
+             overallSuccess:(void (^)())overallSuccess
+                    failure:(void (^)(NSError *error))failure
+{
+    if (mediaObjects.count == 0) {
+        if (overallSuccess) {
+            overallSuccess();
+        }
+        return;
+    }
+
+    NSNumber *totalOperations = @(mediaObjects.count);
+    __block NSUInteger completedOperations = 0;
+
+    void (^individualOperationSuccess)() = ^() {
+        @synchronized (totalOperations) {
+            completedOperations += 1;
+            if (completedOperations >= totalOperations.unsignedIntegerValue && overallSuccess) {
+                overallSuccess();
+            }
+        }
+    };
+
+    for (Media *media in mediaObjects) {
+        // This effectively ignores any errors presented
+        [self updateMedia:media success:individualOperationSuccess failure:individualOperationSuccess];
```

Comment:
Should we ignore the errors at the service level? I prefer to leave that to whom invokes the method. If we have the failure callback we should make use of it.
If not the remove the failure callback and tell in the method comment that all errors are ignored and you callback when all operations finish independently of state.